components/motor/gpiostepper/gpiostepper.go

```diff
@@ -203,37 +206,42 @@ func (m *gpioStepper) doCycle(ctx context.Context) (time.Duration, error) {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
+	// thread waits until something changes the target position in the
+	// gpiostepper struct
 	if m.stepPosition == m.targetStepPosition {
 		return 5 * time.Millisecond, nil
 	}
 
+	// TODO: Setting PWM here works much better than steps to set speed
+	// Redo this part with PWM logic.
 	err := m.doStep(ctx, m.stepPosition < m.targetStepPosition)
 	if err != nil {
 		return time.Second, fmt.Errorf("error stepping %w", err)
 	}
 
-	return time.Duration(int64(time.Microsecond*1000*1000) / int64(math.Abs(float64(m.targetStepsPerSecond)))), nil
+	return m.stepperDelay, nil
 }
 
 // have to be locked to call.
 func (m *gpioStepper) doStep(ctx context.Context, forward bool) error {
-	err := multierr.Combine(
+	if err := multierr.Combine(
 		m.enable(ctx, true),
-		m.stepPin.Set(ctx, true, nil),
 		m.dirPin.Set(ctx, forward, nil),
-	)
-	if err != nil {
+		m.stepPin.Set(ctx, true, nil)); err != nil {
 		return err
 	}
 
-	time.Sleep(time.Duration(m.stepperDelay) * time.Microsecond)
```

Comment:
q: without this sleep, there won't be a sleep between the Set(true) and Set(false). is that intentional? doesn't that mean that false->true->false transition can happen so quickly that the motor won't register the event?

Reply:
Went back and forth. It we were implementing through pwm with a fast rise time, we wouldn't need this logic, but I can confirm that the rise time is janky through my video, so added one sleep between on and off, rather than sleeps of dealy/2 for each. Added a comment explaining if we ever want to explore slightly optimizing where the sleeps are. I'd rather have fewer sleeps than more.